tests/ord-aggregator/tests/handler_test.go

```diff
@@ -0,0 +1,213 @@
+package tests
+
+import (
+	"context"
+	"crypto/tls"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/kyma-incubator/compass/tests/pkg/assertions"
+
+	directorSchema "github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	"github.com/kyma-incubator/compass/tests/pkg/fixtures"
+	"github.com/kyma-incubator/compass/tests/pkg/gql"
+	"github.com/kyma-incubator/compass/tests/pkg/idtokenprovider"
+	"github.com/kyma-incubator/compass/tests/pkg/request"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+	"github.com/tidwall/gjson"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+const (
+	tenantHeader = "Tenant"
+
+	descriptionField      = "value.0.description"
+	shortDescriptionField = "value.0.shortDescription"
+
+	expectedSystemInstanceName        = "test-app"
+	expectedSystemInstanceDescription = "test-app-description"
+	expectedBundleTitle               = "BUNDLE TITLE"
+	expectedBundleDescription         = "lorem ipsum dolor nsq sme"
+	expectedPackageTitle              = "PACKAGE 1 TITLE"
+	expectedPackageDescription        = "lorem ipsum dolor set"
+	expectedProductTitle              = "PRODUCT TITLE"
+	expectedProductShortDescription   = "lorem ipsum"
+	firstAPIExpectedTitle             = "API TITLE"
+	firstAPIExpectedDescription       = "lorem ipsum dolor sit amet"
+	firstEventTitle                   = "EVENT TITLE"
+	firstEventDescription             = "lorem ipsum dolor sit amet"
+	secondEventTitle                  = "EVENT TITLE 2"
+	secondEventDescription            = "lorem ipsum dolor sit amet"
+	expectedTombstoneOrdID            = "ns:apiResource:API_ID2:v1"
+	expectedVendorTitle               = "SAP"
+
+	expectedNumberOfSystemInstances = 1
+	expectedNumberOfPackages        = 1
+	expectedNumberOfBundles         = 1
+	expectedNumberOfProducts        = 1
+	expectedNumberOfAPIs            = 1
+	expectedNumberOfEvents          = 2
+	expectedNumberOfTombstones      = 1
+	expectedNumberOfVendors         = 1
+
+	testTimeoutAdditionalBuffer = 1 * time.Minute
+)
+
+func TestORDAggregator(t *testing.T) {
+	appInput := fixtures.FixSampleApplicationRegisterInputWithORDWebhooks(expectedSystemInstanceName, expectedSystemInstanceDescription, testConfig.ExternalServicesMockBaseURL)
+
+	eventsMap := make(map[string]string, 0)
+	eventsMap[firstEventTitle] = firstEventDescription
+	eventsMap[secondEventTitle] = secondEventDescription
+
+	ctx := context.Background()
+
+	dexToken, err := idtokenprovider.GetDexToken()
```

Comment:
I think that @StanislavStefanov  moved dex token generation to a central place as a test optimization maybe you can look into it.